sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/text/TextIOIT.java

```diff
@@ -0,0 +1,116 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.text;
+
+import com.google.common.base.Function;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.Iterables;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import org.apache.beam.sdk.io.FileSystems;
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.io.TextIO;
+import org.apache.beam.sdk.io.common.HashingFn;
+import org.apache.beam.sdk.io.common.IOTestPipelineOptions;
+import org.apache.beam.sdk.io.common.TestTextLine;
+import org.apache.beam.sdk.io.fs.MatchResult;
+import org.apache.beam.sdk.io.fs.ResourceId;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.Values;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+
+/**
+ * An integration test for {@link org.apache.beam.sdk.io.TextIO}.
+ *
+ * <p>Run this test using the command below. Pass in connection information via PipelineOptions:
+ * <pre>
+ *  mvn -e -Pio-it verify -pl sdks/java/io/text -DintegrationTestPipelineOptions='[
+ *  "--numberOfRecords=100000" ]'
+ * </pre>
+ * */
+@RunWith(JUnit4.class)
+public class TextIOIT {
+
+  private static String filenamePrefix;
+  private static Long linesOfTextCount;
+
+  @Rule
+  public TestPipeline pipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void setup() throws ParseException {
+    PipelineOptionsFactory.register(IOTestPipelineOptions.class);
+    IOTestPipelineOptions options = TestPipeline.testingPipelineOptions()
+        .as(IOTestPipelineOptions.class);
+
+    linesOfTextCount = options.getNumberOfRecords();
+    filenamePrefix = generateFileBasename();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    MatchResult match = Iterables.getOnlyElement(
+        FileSystems.match(Collections.singletonList(String.format("%s*", filenamePrefix))));
+    FileSystems.delete(toResourceIds(match));
+  }
+
+  private Collection<ResourceId> toResourceIds(MatchResult match) throws IOException {
+    return FluentIterable.from(match.metadata())
+        .transform(new Function<MatchResult.Metadata, ResourceId>() {
+          @Override
+          public ResourceId apply(MatchResult.Metadata metadata) {
+            return metadata.resourceId();
+          }
+        }).toList();
+  }
+
+  private static String generateFileBasename() {
+    return String.format("TEXTIOIT_%s", new Date().getTime());
```

Comment:
<!--new_thread; commit:9c483a5dd68450cfc0d8bf1cfdd151961fa6cdaa; resolved:0-->
Looks like this should be moved to a pipeline option to support other file-systems. Can we do that now ?

Reply:
Ok - for now I moved the prefix to PipelineOptions. This may be further extended in the future, eg. when gcs' buckets will be used etc (as described in docs).